src/hotspot/share/opto/mulnode.cpp

```diff
@@ -675,7 +699,7 @@ Node *AndLNode::Ideal(PhaseGVN *phase, bool can_reshape) {
   const jlong mask = t2->get_con();
 
   Node* in1 = in(1);
-  uint op = in1->Opcode();
+  int op = in1->Opcode();
```

Comment:
I got curious and checked the code, we have similar patterns everywhere. Filed [JDK-8278328](https://bugs.openjdk.java.net/browse/JDK-8278328) to clean up this mess.